src/DynamicDiscreteSamplers.jl

```diff
@@ -487,16 +487,17 @@ function set_global_shift_increase!(m::Memory, m2, m3::UInt64, m4) # Increase sh
     weight = UInt64(significand_sum<<shift) + 1
     when is that always 1? when
     UInt64(significand_sum<<shift) == 0
-    and because shift < 0 and significand_sum could be as much as 2^64*2^64/8/8-1 = 2^122-1,
-    shift <= -122
+    and because shift < 0 and significand_sum could be as much as 2^(Base.top_set_bit(m1))*2^64-1,
+    shift <= -Base.top_set_bit(m1)-64
     shift = signed(exponent+m3)
     shift = signed(i-4+m3)
-    signed(i-4+m3) <= -122
-    i <= -signed(m3)-122+4
-    So for -signed(m3)-118 < i, we could need to adjust the ith weight
+    signed(i-4+m3) <= -Base.top_set_bit(m1)-64
+    i <= -signed(m3)-Base.top_set_bit(m1)-64+4
+    So for -signed(m3)-60-Base.top_set_bit(m1) < i, we could need to adjust the ith weight
     =#
-    r0 = max(5, -signed(m3)-117)
-    r1 = m2 # TODO It would be possible to scale this range with length (m[1]) in which case testing could be stricter and performance could be (marginally) better, though not in large cases so possibly not worth doing at all)
+    m1_next2pow = Base.top_set_bit(m[1])
+    r0 = max(5, -signed(m3)-60-m1_next2pow)
```

Comment:
```suggestion
    r0 = max(5, -signed(m3)-60-Base.top_set_bit(m[1]))
```
The variable name `m1_next2pow` is not any easier to understand than the expression `Base.top_set_bit(m[1])` (which is also what is used in the comments).

Reply:
ok, I removed that and put Base.top_set_bit(m[1]) in all places